File: fullstack-online/unit_09/post_it/server/Models/Account.cs
namespace post_it.Models;

public class Account : Profile
{
  // NOTE inheritance!
  // public string Id { get; set; }
  // public string Name { get; set; }
  // public string Picture { get; set; }
  public string Email { get; set; }
}

public class Profile
{
  public string Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public string Name { get; set; }
  public string Picture { get; set; }

}

File: fullstack-online/unit_09/cryptipedia/client/src/models/Account.js
export class Profile {
  constructor(data) {
    this.id = data.id
    this.name = data.name
    this.picture = data.picture
    this.updatedAt = new Date(data.updatedAt)
    this.createdAt = new Date(data.createdAt)
  }
}

export class Account extends Profile {
  /**
   * @typedef AccountData
   * @property {string} id
   * @property {string} email
   * @property {string} name
   * @property {string} picture
   * 
   * @param {AccountData} data
   */
  constructor(data) {
    super(data)
    this.email = data.email
    // TODO add additional properties if needed
  }
}

